### PR TITLE
Codechange: Simplify storage of WaterTileType in map.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Water tile type is not stored directly in the map, it has to under go conversion every time it is looked up,

## Description

Instead of doing the conversion every time `GetWaterTileType()` is called, do the conversion on map load. This replaces the special values with values from the `WaterTileType` instead, and `GetWaterTileType()` becomes a simple accessor.

## Limitations

Needs a savegame bump.

WBL_TYPE_COUNT could maybe be reduced to 2 as only 4 values are used.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/saveload/saveload.h [hunk 1]
```diff
@@ -387,6 +387,7 @@ enum SaveLoadVersion : uint16_t {
 
 	SLV_ROAD_STOP_TILE_DATA,                ///< 340  PR#12883 Move storage of road stop tile data, also save for road waypoints.
 	SLV_COMPANY_ALLOW_LIST_V2,              ///< 341  PR#12908 Fixed savegame format for saving of list of client keys that are allowed to join this company.
+	SLV_WATER_TILE_TYPE,                    ///< 342  PR#..... Simplify water tile type.
```
Comment (rubidium42) [inline src/saveload/saveload.h]:
```suggestion
	SLV_WATER_TILE_TYPE,                    ///< 342  PR#13030 Simplify water tile type.
```

**Review by rubidium42 — COMMENTED** [review]
Looks okay to me, besides the missing PR id.

**Comment by LordAro** [comment]
Any benchmarking?

**Comment by glx22** [comment]
Missing update of html docs I think.

**Review by rubidium42 — APPROVED** [review]

**Comment by PeterN** [comment]
> Any benchmarking?

Did not do any. I should think that (once converted) there is almost no difference with optimisations enabled.

**Pull request merged**